tests/test_migrations.py

```diff
@@ -72,109 +71,122 @@ def test_migrations_and_tasks(tmp_path: Path):
     assert (dst / "PEP440-1.0.0-2-2.0-before.json").is_file()
     assert (dst / "PEP440-1.0.0-2-2.0-after.json").is_file()
     answers = yaml.safe_load((dst / ".copier-answers.yml").read_text())
-    assert answers == {"_commit": "v2.0", "_src_path": str(git_src)}
+    assert answers == {"_commit": "v2.0", "_src_path": str(src)}
 
 
-def test_pre_migration_modifies_answers(tmp_path_factory):
+def test_pre_migration_modifies_answers(
+    tmp_path_factory: pytest.TempPathFactory,
+) -> None:
     """Test support for answers modifications in pre-migrations."""
-    template = tmp_path_factory.mktemp("template")
-    subproject = tmp_path_factory.mktemp("subproject")
+    src, dst = map(tmp_path_factory.mktemp, ("src", "dst"))
+
     # v1 of template asks for a favourite song and writes it to songs.json
-    with local.cwd(template):
-        build_file_tree(
-            {
-                "[[ _copier_conf.answers_file ]].jinja": "[[ _copier_answers|tojson ]]",
-                "copier.yml": f"""\
-                    _envops: {BRACKET_ENVOPS_JSON}
-                    best_song: la vie en rose
-                    """,
-                "songs.json.jinja": "[ [[ best_song|tojson ]] ]",
-            }
-        )
+    build_file_tree(
+        {
+            (src / "[[ _copier_conf.answers_file ]].jinja"): (
+                "[[ _copier_answers|tojson ]]"
+            ),
+            (src / "copier.yml"): (
+                f"""\
+                _envops: {BRACKET_ENVOPS_JSON}
+                best_song: la vie en rose
+                """
+            ),
+            (src / "songs.json.jinja"): "[ [[ best_song|tojson ]] ]",
+        }
+    )
+    with local.cwd(src):
         git("init")
         git("add", ".")
         git("commit", "-m1")
         git("tag", "v1")
+
     # User copies v1 template into subproject
-    with local.cwd(subproject):
-        copy(src_path=str(template), defaults=True, overwrite=True)
-        answers = json.loads(Path(".copier-answers.yml").read_text())
-        assert answers["_commit"] == "v1"
-        assert answers["best_song"] == "la vie en rose"
-        assert json.loads(Path("songs.json").read_text()) == ["la vie en rose"]
+    copy(src_path=str(src), dst_path=dst, defaults=True, overwrite=True)
+    answers = json.loads((dst / ".copier-answers.yml").read_text())
+    assert answers["_commit"] == "v1"
+    assert answers["best_song"] == "la vie en rose"
+    assert json.loads((dst / "songs.json").read_text()) == ["la vie en rose"]
```

Comment:
With this change, you're altering the test.

The previous implementation made sure that `dst_path` defaults to `.`, where the new one doesn't.

Small but important details. Please undo these kind of changes.

Reply:
Ouch, I see and hadn't thought about that! Good catch! :monocle_face::bow:

But one remark: I understand that my change did change the exact test scenario. But I think testing that `dst_path` defaults to `.` should be covered in a dedicated test (which I don't see at the moment) and not only implicitly in some unrelated test because then a change like mine, which was perfectly fine w.r.t. to the scenario under test, coincidentally removed another implicit test. WDYT, @yajo?

Reply:
Of course, specific tests are better. But as long as we don't touch them, they won't add too much. Just undo the change; that'll be the most practical solution.